### PR TITLE
Provide a simple way to run locally

### PR DESCRIPTION
**Pull request by toch**
See https://bluesquare.atlassian.net/browse/HEXA-980

This a first step to run openhexa locally as a linux service.

Could you check the environment variables (especially the commented ones) and tell me if I can indeed remove them?

### CODE REVIEW

File: openhexa.service [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Description=%i service
+PartOf=docker.service network-online.target
+After=docker.service network-online.target
+AssertPathExists=/usr/local/openhexa/compose.yml
+
+[Service]
+Type=oneshot
+RemainAfterExit=true
+WorkingDirectory=/etc/openhexa
+ExecStart=/usr/bin/docker compose --file /usr/local/openhexa/compose.yml --project-name openhexa --profile frontend --profile minio --profile pipelines up -d --remove-orphans
```
Comment (qgerome) [inline openhexa.service:11]: Shouldn't we use `start` instead of `up` to avoid to use newer images by inadvertance?

**Comment by toch** [comment]
I've finished to setup the debian packaging. Here the result:

```
❯ sudo dpkg -i openhexa_1.0-1_amd64.deb                                                                                                                                                     
Selecting previously unselected package openhexa.
(Reading database ... 429409 files and directories currently installed.)
Preparing to unpack openhexa_1.0-1_amd64.deb ...
Unpacking openhexa (1.0-1) ...
Setting up openhexa (1.0-1) ...
WARN[0000] /usr/share/openhexa/compose.yml: `version` is obsolete 
[+] Pulling 6/6
 ✔ pipelines_runner Skipped - Image is already being pulled by app                                                                                                                                          0.0s 
 ✔ pipelines_scheduler Skipped - Image is already being pulled by app                                                                                                                                       0.0s 
 ✔ app Pulled                                                                                                                                                                                               1.3s 
 ✔ frontend Pulled                                                                                                                                                                                          1.3s 
 ✔ minio Pulled                                                                                                                                                                                             1.5s 
 ✔ db Pulled                                                                                                                                                                                                1.3s 
Created symlink /etc/systemd/system/multi-user.target.wants/openhexa.service → /lib/systemd/system/openhexa.service.
❯ systemctl status openhexa                                                                                                                                                                                      
● openhexa.service -  service
     Loaded: loaded (/lib/systemd/system/openhexa.service; enabled; preset: enabled)
     Active: active (exited) since Tue 2024-07-02 16:32:41 CEST; 14s ago
    Process: 164084 ExecStart=/usr/bin/docker compose --file /usr/share/openhexa/compose.yml --env-file /etc/openhexa/env.conf --project-name openhexa --profile frontend --profile minio --profile pipelines up>
   Main PID: 164084 (code=exited, status=0/SUCCESS)
        CPU: 403ms

jui 02 16:32:40 kisuke docker[164106]:  Container openhexa-db-1  Started
jui 02 16:32:40 kisuke docker[164106]:  Container openhexa-pipelines_runner-1  Starting
jui 02 16:32:40 kisuke docker[164106]:  Container openhexa-pipelines_scheduler-1  Starting
jui 02 16:32:40 kisuke docker[164106]:  Container app  Starting
jui 02 16:32:40 kisuke docker[164106]:  Container openhexa-pipelines_scheduler-1  Started
jui 02 16:32:41 kisuke docker[164106]:  Container openhexa-pipelines_runner-1  Started
jui 02 16:32:41 kisuke docker[164106]:  Container app  Started
jui 02 16:32:41 kisuke docker[164106]:  Container frontend  Starting
```

File: README.md [hunk 2]
```diff
@@ -44,3 +44,44 @@ Please report bugs in the issues section of this repository: https://github.com/
...
+
+```bash
+sudo dpkg -i openhexa_1.0-1.deb
+```
+
+It installs the systemd service `openhexa`, that you can manage with `systemctl`.
+
+#### Configure
+
+You can change some config via the file `/etc/openhexa.conf`.
```
Comment (qgerome) [inline README.md:87]: I didn't see this file in the package, is it still used or need to be implemented ?

**Review by qgerome — COMMENTED** [review]
Great!
I don't have a lot to say about the code in itself because I lack the expertise but I have more global questions on the "what's next?":
* How do you configure the setup, the ports, ... ?
* Do you think we have to package the DB as well? (I'd assume we do not)
* There is still a step of initial configuration of the app that is needed, do you think it's to be handled via this script ?

File: README.md (hunk 2 above)
Comment (toch) [reply README.md:87]: It's a typo, it should be `/etc/openhexa/env.conf`. Thanks

**Comment by toch** [comment]
> * How do you configure the setup, the ports, ... ?

The setup can be done with the file `/etc/openhexa/env.conf`. You can set multiple environment variables, including new ones I've added to set the published ports for the app, frontend, db, and minio.

> * Do you think we have to package the DB as well? (I'd assume we do not)

I'd like to rely on existing debian packages for the third party services such as the db or even minio (they provide one `.deb`, and until we support local storage).

I haven't done it for now, as I like to do one change at a time.

> * There is still a step of initial configuration of the app that is needed, do you think it's to be handled via this script ?

we can. could you tell me what config needs to be done?

**Comment by qgerome** [comment]
> > * Do you think we have to package the DB as well? (I'd assume we do not)
> 
> I'd like to rely on existing debian packages for the third party services such as the db or even minio (they provide one `.deb`, and until we support local storage).
> 
> I haven't done it for now, as I like to do one change at a time.

👌🏻 

> > * There is still a step of initial configuration of the app that is needed, do you think it's to be handled via this script ?
> 
> we can. could you tell me what config needs to be done?

I'm talking about this step (creation of the root user, migrating the DB, loading default fixtures)
https://github.com/BLSQ/openhexa-app/blob/a0f89d5da1717318bd1dd6eed366e64724027dd0/docker-entrypoint.sh#L73

**Comment by toch** [comment]
> > > * There is still a step of initial configuration of the app that is needed, do you think it's to be handled via this script ?
> > 
> > 
> > we can. could you tell me what config needs to be done?
> 
> I'm talking about this step (creation of the root user, migrating the DB, loading default fixtures) https://github.com/BLSQ/openhexa-app/blob/a0f89d5da1717318bd1dd6eed366e64724027dd0/docker-entrypoint.sh#L73

that's definitively something I can do in the next step (that also takes care of the upgrade)

**Comment by qgerome** [comment]
> > > > * There is still a step of initial configuration of the app that is needed, do you think it's to be handled via this script ?
> > > 
> > > 
> > > we can. could you tell me what config needs to be done?
> > 
> > 
> > I'm talking about this step (creation of the root user, migrating the DB, loading default fixtures) https://github.com/BLSQ/openhexa-app/blob/a0f89d5da1717318bd1dd6eed366e64724027dd0/docker-entrypoint.sh#L73
> 
> that's definitively something I can do in the next step (that also takes care of the upgrade)
> 
> are the fixtures really needed?

The fixtures contains the feature flags instances. We could also replace the fixtures by a DB migration if needed.

**Comment by toch** [comment]
Thanks, [ticket #HEXA-983 updated accordingly](https://bluesquare.atlassian.net/browse/HEXA-983)

To extract the db and make it a debian package dep, here the [new ticket #HEXA-1010](https://bluesquare.atlassian.net/browse/HEXA-1010).

**Comment by qgerome** [comment]
Do you prefer to merge this PR and work in a new one or work in this one?

**Comment by toch** [comment]
> Do you prefer to merge this PR and work in a new one or work in this one?

I was going to merge this PR as soon I've fixed the typo in the doc. Is that ok for you? It seems to me it's a usable step and not too big.

**Review by nazarfil — APPROVED** [review]

**Pull request merged**